### PR TITLE
Add bus event consumers

### PR DESCRIPTION
**Pull request by Hatleskog**
Add topic and queue consumers for PI entities

### CODE REVIEW

File: src/Equinor.ProCoSys.Completion.WebApi/Synchronization/DocumentEventConsumer.cs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    {
+        var busEvent = context.Message;
+
+        ValidateMessage(busEvent);
+        _plantSetter.SetPlant(busEvent.Plant);
+
+        if (await _documentRepository.ExistsAsync(busEvent.ProCoSysGuid, context.CancellationToken))
+        {
+            // TODO Implement mapping ?
+            var document = await _documentRepository.GetAsync(busEvent.ProCoSysGuid, context.CancellationToken);
```
[the remark this answers is not in the document]
Comment (TordJoranger) [reply src/Equinor.ProCoSys.Completion.WebApi/Synchronization/DocumentEventConsumer.cs:47]: also remove commendted code ;)

File: src/Equinor.ProCoSys.Completion.WebApi/Synchronization/DocumentEventConsumer.cs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+            documentEvent.Plant,
+            documentEvent.ProCoSysGuid,
+            documentEvent.DocumentNo
+        );
+        return document;
+    }
+
+    
+    private static void MapFromEventToDocument(IDocumentEventV1 documentEvent, Document document)
+    {
```
Comment (TordJoranger) [inline src/Equinor.ProCoSys.Completion.WebApi/Synchronization/DocumentEventConsumer.cs:88]: remove

File: src/Equinor.ProCoSys.Completion.WebApi/Synchronization/LibraryEventConsumer.cs [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+            libraryEvent.Description!,
+            !Enum.TryParse(libraryEvent.Type, true, out LibraryType libType)
+                && libraryEvent.Type.ToUpper() == "COMM_PRIORITY"
+                ? LibraryType.PUNCHLIST_PRIORITY : libType
+            );
+        return library;
+    }
+  
+    private static void MapFromEventToLibrary(ILibraryEventV1 libraryEvent, LibraryItem library)
+    {
```
Comment (TordJoranger) [inline src/Equinor.ProCoSys.Completion.WebApi/Synchronization/LibraryEventConsumer.cs:108]: I think we need to map all fields that we keep in the DB, to keep things in sync.

File: src/Equinor.ProCoSys.Completion.WebApi/Synchronization/PunchItemEventConsumer.cs [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+        {
+            throw new Exception("Message is missing ProCoSysGuid");
+        }
+
+        if (string.IsNullOrEmpty(busEvent.Plant))
+        {
+            throw new Exception("Message is missing Plant");
+        }
+
+        if (string.IsNullOrEmpty(busEvent.Description))
```
Comment (TordJoranger) [inline src/Equinor.ProCoSys.Completion.WebApi/Synchronization/PunchItemEventConsumer.cs:96]: Description required? (just checking)

File: src/Equinor.ProCoSys.Completion.WebApi/Synchronization/PunchItemEventConsumer.cs [hunk 5]
```diff
@@ -0,0 +1,340 @@
...
+
+        SetExternalItemNo(punchItem, busEvent.ExternalItemNo);
+        SetMaterialRequired(punchItem, busEvent.MaterialRequired);
+        SetMaterialETAUtc(punchItem, busEvent.MaterialETA);
+        SetMaterialExternalNo(punchItem, busEvent.MaterialExternalNo);
+    }
+
+    private void SetMaterialExternalNo(PunchItem punchItem, string? materialExternalNo)
+    {
+        if (materialExternalNo is null)
```
Comment (TordJoranger) [inline src/Equinor.ProCoSys.Completion.WebApi/Synchronization/PunchItemEventConsumer.cs:155]: Are theese correct. what if I remove materialExternalNo in Pcs4?

File: src/Equinor.ProCoSys.Completion.WebApi/Synchronization/PunchItemEventConsumer.cs (hunk 4 above)
Comment (Hatleskog) [reply src/Equinor.ProCoSys.Completion.WebApi/Synchronization/PunchItemEventConsumer.cs:96]: not null i db ja.

**Review by TordJoranger — APPROVED** [review]

**Pull request merged**